docs/contrib/bindgen.rst

```diff
@@ -0,0 +1,40 @@
+Automatic binding generation
+============================
+
+Binding generation for now is experimental. The goal is to first use it
+internally to generate bindings for C's stdlib and later make it
+available via scala.meta macros and sbt tasks.
```

Comment:
I would vote for code generation here. The less powerful technique that works well with IDEs